go/proxyd/latestblock.go

```diff
@@ -0,0 +1,61 @@
+package proxyd
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type LatestBlockHead struct {
+	url  string
+	quit chan struct{}
+
+	mutex sync.Mutex
```

Comment:
Any reason to use a `sync.Mutex` over a `sync.RWMutex`?

Reply:
Only to keep it simple. I don't have any strong opinions on either so I can make the change.